From b677148b1fde36ade7ac2c4529b644cbd975e76b Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 16 Jan 2024 16:46:41 +0200
Subject: [PATCH] [SW-170792] gaudi3: pif: don't turn mctp msg to inbound

As a reset default MCTP messages are turned to an inbound transactions
to the NOC.
Since at the moment those transactions are not handled and not expected,
this translation is turned off.
In addition, the MCTP Queue is getting overfilled because the MCTP msgs
are not handled. this event is masked for now.

Change-Id: I43639253c56ccad09d02e0feb9f9313687ff4fca
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index f549539..e4b8de9 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -32,6 +32,8 @@
 #include <pcie_phy_regs.h>
 #include <pcie_wrap_regs.h>
 #include <pcie_wrap_masks.h>
+#include <pcie_wrap_mctp_masks.h>
+#include <pcie_wrap_mctp_regs.h>
 #include <arc_regs.h>
 #include <pll_ctrl_regs.h>
 #include <gaudi3_blocks_linux_driver.h>
@@ -1023,6 +1025,18 @@
 			PCIE_DBI_PCI_MSIX_CAP_ID_NEXT_CTRL_REG_PCI_MSIX_CAP_NEXT_OFFSET_M);
 }
 
+#define MCTP_QUEUE_DROP_EVT_MASK	BIT(2)
+
+static void pif_arc_config_mctp_params(void)
+{
+	/* mask HBW MCTP Queue Drop event */
+	regmap_writel(MCTP_QUEUE_DROP_EVT_MASK, mmD0_PCIE_WRAP_MCTP_BASE +
+						mmPCIE_WRAP_MCTP_INTR_MASK);
+	/* disable turning mctp msg to inbound transaction */
+	regmap_writel(PCIE_WRAP_MCTP_MCTP_IN_ENABLE_V_M, mmD0_PCIE_WRAP_MCTP_BASE +
+							 mmPCIE_WRAP_MCTP_MCTP_IN_ENABLE);
+}
+
 __weak void preboot_update_bar_sizes(void) {}
 __weak void pcie_wait_for_sram_ext_ld_done(void) {}
 
@@ -1037,6 +1051,8 @@
 
 	pif_arc_config_dbi_params();
 
+	pif_arc_config_mctp_params();
+
 	/* Set HBM BAR size */
 	preboot_update_bar_sizes();
 
